controllers/talosconfig_controller.go

```diff
@@ -80,16 +86,51 @@ type talosConfigContext struct {
 	Key    string
 }
 
-func (r *TalosConfigReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		WithOptions(options).
+func (r *TalosConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	r.Scheme = mgr.GetScheme()
+
+	b := ctrl.NewControllerManagedBy(mgr).
 		For(&bootstrapv1alpha3.TalosConfig{}).
-		Complete(r)
+		WithOptions(options).
+		Watches(
+			&source.Kind{Type: &capiv1.Machine{}},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: handler.ToRequestsFunc(r.MachineToBootstrapMapFunc),
```

Comment:
dumb question: how did our provider work before? if it didn't have this watch?

Reply:
If i understand it correctly myself, this only comes into play when you add the extra "watches" to the controller. By default, it owns "TalosConfig" so would only trigger a reconcile when that resource changed. But adding watches for Machines and MachinePools requires the handler functions. 

We also do something similar in the controlplane provider where we have extra watchers for cluster resource.

Reply:
yep, I understand that, I mean if watch for Machine is required for the provider to function properly, how did it work before this PR? as it doesn't watch Machine?